test/test_torch.py

```diff
@@ -13451,8 +13326,126 @@ def add_neg_dim_tests():
         assert not hasattr(_TestTorchMixin, test_name), "Duplicated test name: " + test_name
         setattr(_TestTorchMixin, test_name, make_neg_dim_test(name, tensor_arg, arg_constr, types, extra_dim))
 
-add_neg_dim_tests()
+# Device-agnostic tests. Instantiated below and not run directly.
+class TestTorchDeviceType(TestCase):
+    def test_diagonal(self, device):
+        x = torch.randn((100, 100), device=device)
+        result = torch.diagonal(x)
+        expected = torch.diag(x)
+        self.assertEqual(result, expected)
+
+        x = torch.randn((100, 100), device=device)
+        result = torch.diagonal(x, 17)
+        expected = torch.diag(x, 17)
+        self.assertEqual(result, expected)
+
+    def test_neg(self, device):
+        int_types = [torch.int, torch.short, torch.int8, torch.uint8]
+        float_types = [torch.float, torch.double, torch.long]
```

Comment:
@gchanan There's still an explicit list here!

Reply:
also interesting that `float_types` has `long` in the list?

Reply:
which runs randn?  how can this possibly work?